crates/driver/src/boundary/liquidity/balancer/v2/mod.rs

```diff
@@ -75,19 +76,22 @@ fn to_interaction(
 }
 
 pub fn collector(
+    graph_api_base_url: &Url,
     eth: &Ethereum,
     block_stream: CurrentBlockStream,
     block_retriever: Arc<dyn BlockRetrieving>,
     config: &infra::liquidity::config::BalancerV2,
 ) -> Box<dyn LiquidityCollecting> {
     let eth = Arc::new(eth.clone());
+    let graph_api_base_url = Arc::new(graph_api_base_url.clone());
```

Comment:
What is the point of using an Arc here, if `init_liquidity` doesn't take a shared ref?

Reply:
otherwise it complains that "borrowed data escapes outside of function"